src/impl/sctptransport.cpp

```diff
@@ -284,6 +284,16 @@ SctpTransport::SctpTransport(shared_ptr<Transport> lower, const Configuration &c
 		throw std::runtime_error("Could not disable SCTP fragmented interleave, errno=" +
 		                         std::to_string(errno));
 
+	// When using SCTP over DTLS, the data integrity is ensured by DTLS, so there's no need to
+	// additionally compute CRC32c.
+	// See https://datatracker.ietf.org/doc/html/draft-ietf-tsvwg-sctp-zero-checksum
+	if (config.sctpZeroChecksum) {
+		int on = 1;
+		if (usrsctp_setsockopt(mSock, IPPROTO_SCTP, SCTP_ACCEPT_ZERO_CHECKSUM, &on, sizeof(on)))
```

Comment:
I don't think the parameter does what you expect: it doesn't disable checksums for outgoing packets, it only tells the remote side that this side is willing to accept zero-checksum packets.

Reply:
Additionally, the expected parameter is not a boolean, the value should be instead set to `SCTP_EDMID_LOWER_LAYER_DTLS` as indicated [in the draft](https://datatracker.ietf.org/doc/html/draft-ietf-tsvwg-sctp-zero-checksum#section-7.1).